diff -Naur gen-java/edu/berkeley/cs/amplab/carat/thrift/BatteryDetails.java thrift/edu/berkeley/cs/amplab/carat/thrift/BatteryDetails.java
--- gen-java/edu/berkeley/cs/amplab/carat/thrift/BatteryDetails.java	2012-05-02 17:42:27.554947959 -0700
+++ thrift/edu/berkeley/cs/amplab/carat/thrift/BatteryDetails.java	2012-05-02 21:32:08.547011150 -0700
@@ -24,8 +24,6 @@
 import java.util.BitSet;
 import java.nio.ByteBuffer;
 import java.util.Arrays;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 public class BatteryDetails implements org.apache.thrift.TBase<BatteryDetails, BatteryDetails._Fields>, java.io.Serializable, Cloneable {
   private static final org.apache.thrift.protocol.TStruct STRUCT_DESC = new org.apache.thrift.protocol.TStruct("BatteryDetails");
@@ -579,7 +577,7 @@
     try {
       write(new org.apache.thrift.protocol.TCompactProtocol(new org.apache.thrift.transport.TIOStreamTransport(out)));
     } catch (org.apache.thrift.TException te) {
-      throw new java.io.IOException(te);
+      throw new java.io.IOException(te.getMessage());
     }
   }
 
@@ -589,7 +587,7 @@
       __isset_bit_vector = new BitSet(1);
       read(new org.apache.thrift.protocol.TCompactProtocol(new org.apache.thrift.transport.TIOStreamTransport(in)));
     } catch (org.apache.thrift.TException te) {
-      throw new java.io.IOException(te);
+      throw new java.io.IOException(te.getMessage());
     }
   }
 
diff -Naur gen-java/edu/berkeley/cs/amplab/carat/thrift/CallInfo.java thrift/edu/berkeley/cs/amplab/carat/thrift/CallInfo.java
--- gen-java/edu/berkeley/cs/amplab/carat/thrift/CallInfo.java	2012-05-02 17:42:27.558947981 -0700
+++ thrift/edu/berkeley/cs/amplab/carat/thrift/CallInfo.java	2012-05-02 21:32:08.547011150 -0700
@@ -24,8 +24,6 @@
 import java.util.BitSet;
 import java.nio.ByteBuffer;
 import java.util.Arrays;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 public class CallInfo implements org.apache.thrift.TBase<CallInfo, CallInfo._Fields>, java.io.Serializable, Cloneable {
   private static final org.apache.thrift.protocol.TStruct STRUCT_DESC = new org.apache.thrift.protocol.TStruct("CallInfo");
@@ -497,7 +495,7 @@
     try {
       write(new org.apache.thrift.protocol.TCompactProtocol(new org.apache.thrift.transport.TIOStreamTransport(out)));
     } catch (org.apache.thrift.TException te) {
-      throw new java.io.IOException(te);
+      throw new java.io.IOException(te.getMessage());
     }
   }
 
@@ -507,7 +505,7 @@
       __isset_bit_vector = new BitSet(1);
       read(new org.apache.thrift.protocol.TCompactProtocol(new org.apache.thrift.transport.TIOStreamTransport(in)));
     } catch (org.apache.thrift.TException te) {
-      throw new java.io.IOException(te);
+      throw new java.io.IOException(te.getMessage());
     }
   }
 
diff -Naur gen-java/edu/berkeley/cs/amplab/carat/thrift/CaratService.java thrift/edu/berkeley/cs/amplab/carat/thrift/CaratService.java
--- gen-java/edu/berkeley/cs/amplab/carat/thrift/CaratService.java	2012-05-02 17:42:27.598948176 -0700
+++ thrift/edu/berkeley/cs/amplab/carat/thrift/CaratService.java	2012-05-02 21:32:08.543011135 -0700
@@ -24,8 +24,6 @@
 import java.util.BitSet;
 import java.nio.ByteBuffer;
 import java.util.Arrays;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 public class CaratService {
 
@@ -310,7 +308,7 @@
   }
 
   public static class Processor<I extends Iface> extends org.apache.thrift.TBaseProcessor<I> implements org.apache.thrift.TProcessor {
-    private static final Logger LOGGER = LoggerFactory.getLogger(Processor.class.getName());
+    //private static final Logger LOGGER = LoggerFactory.getLogger(Processor.class.getName());
     public Processor(I iface) {
       super(iface, getProcessMap(new HashMap<String, org.apache.thrift.ProcessFunction<I, ? extends org.apache.thrift.TBase>>()));
     }
@@ -648,7 +646,7 @@
       try {
         write(new org.apache.thrift.protocol.TCompactProtocol(new org.apache.thrift.transport.TIOStreamTransport(out)));
       } catch (org.apache.thrift.TException te) {
-        throw new java.io.IOException(te);
+        throw new java.io.IOException(te.getMessage());
       }
     }
 
@@ -656,7 +654,7 @@
       try {
         read(new org.apache.thrift.protocol.TCompactProtocol(new org.apache.thrift.transport.TIOStreamTransport(in)));
       } catch (org.apache.thrift.TException te) {
-        throw new java.io.IOException(te);
+        throw new java.io.IOException(te.getMessage());
       }
     }
 
@@ -1003,7 +1001,7 @@
       try {
         write(new org.apache.thrift.protocol.TCompactProtocol(new org.apache.thrift.transport.TIOStreamTransport(out)));
       } catch (org.apache.thrift.TException te) {
-        throw new java.io.IOException(te);
+        throw new java.io.IOException(te.getMessage());
       }
     }
 
@@ -1011,7 +1009,7 @@
       try {
         read(new org.apache.thrift.protocol.TCompactProtocol(new org.apache.thrift.transport.TIOStreamTransport(in)));
       } catch (org.apache.thrift.TException te) {
-        throw new java.io.IOException(te);
+        throw new java.io.IOException(te.getMessage());
       }
     }
 
@@ -1357,7 +1355,7 @@
       try {
         write(new org.apache.thrift.protocol.TCompactProtocol(new org.apache.thrift.transport.TIOStreamTransport(out)));
       } catch (org.apache.thrift.TException te) {
-        throw new java.io.IOException(te);
+        throw new java.io.IOException(te.getMessage());
       }
     }
 
@@ -1365,7 +1363,7 @@
       try {
         read(new org.apache.thrift.protocol.TCompactProtocol(new org.apache.thrift.transport.TIOStreamTransport(in)));
       } catch (org.apache.thrift.TException te) {
-        throw new java.io.IOException(te);
+        throw new java.io.IOException(te.getMessage());
       }
     }
 
@@ -1800,7 +1798,7 @@
       try {
         write(new org.apache.thrift.protocol.TCompactProtocol(new org.apache.thrift.transport.TIOStreamTransport(out)));
       } catch (org.apache.thrift.TException te) {
-        throw new java.io.IOException(te);
+        throw new java.io.IOException(te.getMessage());
       }
     }
 
@@ -1808,7 +1806,7 @@
       try {
         read(new org.apache.thrift.protocol.TCompactProtocol(new org.apache.thrift.transport.TIOStreamTransport(in)));
       } catch (org.apache.thrift.TException te) {
-        throw new java.io.IOException(te);
+        throw new java.io.IOException(te.getMessage());
       }
     }
 
@@ -2210,7 +2208,7 @@
       try {
         write(new org.apache.thrift.protocol.TCompactProtocol(new org.apache.thrift.transport.TIOStreamTransport(out)));
       } catch (org.apache.thrift.TException te) {
-        throw new java.io.IOException(te);
+        throw new java.io.IOException(te.getMessage());
       }
     }
 
@@ -2218,7 +2216,7 @@
       try {
         read(new org.apache.thrift.protocol.TCompactProtocol(new org.apache.thrift.transport.TIOStreamTransport(in)));
       } catch (org.apache.thrift.TException te) {
-        throw new java.io.IOException(te);
+        throw new java.io.IOException(te.getMessage());
       }
     }
 
@@ -2657,7 +2655,7 @@
       try {
         write(new org.apache.thrift.protocol.TCompactProtocol(new org.apache.thrift.transport.TIOStreamTransport(out)));
       } catch (org.apache.thrift.TException te) {
-        throw new java.io.IOException(te);
+        throw new java.io.IOException(te.getMessage());
       }
     }
 
@@ -2665,7 +2663,7 @@
       try {
         read(new org.apache.thrift.protocol.TCompactProtocol(new org.apache.thrift.transport.TIOStreamTransport(in)));
       } catch (org.apache.thrift.TException te) {
-        throw new java.io.IOException(te);
+        throw new java.io.IOException(te.getMessage());
       }
     }
 
@@ -3067,7 +3065,7 @@
       try {
         write(new org.apache.thrift.protocol.TCompactProtocol(new org.apache.thrift.transport.TIOStreamTransport(out)));
       } catch (org.apache.thrift.TException te) {
-        throw new java.io.IOException(te);
+        throw new java.io.IOException(te.getMessage());
       }
     }
 
@@ -3075,7 +3073,7 @@
       try {
         read(new org.apache.thrift.protocol.TCompactProtocol(new org.apache.thrift.transport.TIOStreamTransport(in)));
       } catch (org.apache.thrift.TException te) {
-        throw new java.io.IOException(te);
+        throw new java.io.IOException(te.getMessage());
       }
     }
 
diff -Naur gen-java/edu/berkeley/cs/amplab/carat/thrift/CpuStatus.java thrift/edu/berkeley/cs/amplab/carat/thrift/CpuStatus.java
--- gen-java/edu/berkeley/cs/amplab/carat/thrift/CpuStatus.java	2012-05-02 17:42:27.554947959 -0700
+++ thrift/edu/berkeley/cs/amplab/carat/thrift/CpuStatus.java	2012-05-02 21:32:08.543011135 -0700
@@ -24,8 +24,6 @@
 import java.util.BitSet;
 import java.nio.ByteBuffer;
 import java.util.Arrays;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 public class CpuStatus implements org.apache.thrift.TBase<CpuStatus, CpuStatus._Fields>, java.io.Serializable, Cloneable {
   private static final org.apache.thrift.protocol.TStruct STRUCT_DESC = new org.apache.thrift.protocol.TStruct("CpuStatus");
@@ -348,7 +346,7 @@
     try {
       write(new org.apache.thrift.protocol.TCompactProtocol(new org.apache.thrift.transport.TIOStreamTransport(out)));
     } catch (org.apache.thrift.TException te) {
-      throw new java.io.IOException(te);
+      throw new java.io.IOException(te.getMessage());
     }
   }
 
@@ -358,7 +356,7 @@
       __isset_bit_vector = new BitSet(1);
       read(new org.apache.thrift.protocol.TCompactProtocol(new org.apache.thrift.transport.TIOStreamTransport(in)));
     } catch (org.apache.thrift.TException te) {
-      throw new java.io.IOException(te);
+      throw new java.io.IOException(te.getMessage());
     }
   }
 
diff -Naur gen-java/edu/berkeley/cs/amplab/carat/thrift/DetailScreenReport.java thrift/edu/berkeley/cs/amplab/carat/thrift/DetailScreenReport.java
--- gen-java/edu/berkeley/cs/amplab/carat/thrift/DetailScreenReport.java	2012-05-02 17:42:27.570948040 -0700
+++ thrift/edu/berkeley/cs/amplab/carat/thrift/DetailScreenReport.java	2012-05-02 21:32:08.543011135 -0700
@@ -24,8 +24,6 @@
 import java.util.BitSet;
 import java.nio.ByteBuffer;
 import java.util.Arrays;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 public class DetailScreenReport implements org.apache.thrift.TBase<DetailScreenReport, DetailScreenReport._Fields>, java.io.Serializable, Cloneable {
   private static final org.apache.thrift.protocol.TStruct STRUCT_DESC = new org.apache.thrift.protocol.TStruct("DetailScreenReport");
@@ -542,7 +540,7 @@
     try {
       write(new org.apache.thrift.protocol.TCompactProtocol(new org.apache.thrift.transport.TIOStreamTransport(out)));
     } catch (org.apache.thrift.TException te) {
-      throw new java.io.IOException(te);
+      throw new java.io.IOException(te.getMessage());
     }
   }
 
@@ -552,7 +550,7 @@
       __isset_bit_vector = new BitSet(1);
       read(new org.apache.thrift.protocol.TCompactProtocol(new org.apache.thrift.transport.TIOStreamTransport(in)));
     } catch (org.apache.thrift.TException te) {
-      throw new java.io.IOException(te);
+      throw new java.io.IOException(te.getMessage());
     }
   }
 
diff -Naur gen-java/edu/berkeley/cs/amplab/carat/thrift/Feature.java thrift/edu/berkeley/cs/amplab/carat/thrift/Feature.java
--- gen-java/edu/berkeley/cs/amplab/carat/thrift/Feature.java	2012-05-02 17:42:27.586948120 -0700
+++ thrift/edu/berkeley/cs/amplab/carat/thrift/Feature.java	2012-05-02 21:32:08.547011150 -0700
@@ -24,8 +24,6 @@
 import java.util.BitSet;
 import java.nio.ByteBuffer;
 import java.util.Arrays;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 public class Feature implements org.apache.thrift.TBase<Feature, Feature._Fields>, java.io.Serializable, Cloneable {
   private static final org.apache.thrift.protocol.TStruct STRUCT_DESC = new org.apache.thrift.protocol.TStruct("Feature");
@@ -355,7 +353,7 @@
     try {
       write(new org.apache.thrift.protocol.TCompactProtocol(new org.apache.thrift.transport.TIOStreamTransport(out)));
     } catch (org.apache.thrift.TException te) {
-      throw new java.io.IOException(te);
+      throw new java.io.IOException(te.getMessage());
     }
   }
 
@@ -363,7 +361,7 @@
     try {
       read(new org.apache.thrift.protocol.TCompactProtocol(new org.apache.thrift.transport.TIOStreamTransport(in)));
     } catch (org.apache.thrift.TException te) {
-      throw new java.io.IOException(te);
+      throw new java.io.IOException(te.getMessage());
     }
   }
 
diff -Naur gen-java/edu/berkeley/cs/amplab/carat/thrift/HogBugReport.java thrift/edu/berkeley/cs/amplab/carat/thrift/HogBugReport.java
--- gen-java/edu/berkeley/cs/amplab/carat/thrift/HogBugReport.java	2012-05-02 17:42:27.586948120 -0700
+++ thrift/edu/berkeley/cs/amplab/carat/thrift/HogBugReport.java	2012-05-02 21:32:08.547011150 -0700
@@ -24,8 +24,6 @@
 import java.util.BitSet;
 import java.nio.ByteBuffer;
 import java.util.Arrays;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 public class HogBugReport implements org.apache.thrift.TBase<HogBugReport, HogBugReport._Fields>, java.io.Serializable, Cloneable {
   private static final org.apache.thrift.protocol.TStruct STRUCT_DESC = new org.apache.thrift.protocol.TStruct("HogBugReport");
@@ -378,7 +376,7 @@
     try {
       write(new org.apache.thrift.protocol.TCompactProtocol(new org.apache.thrift.transport.TIOStreamTransport(out)));
     } catch (org.apache.thrift.TException te) {
-      throw new java.io.IOException(te);
+      throw new java.io.IOException(te.getMessage());
     }
   }
 
@@ -386,7 +384,7 @@
     try {
       read(new org.apache.thrift.protocol.TCompactProtocol(new org.apache.thrift.transport.TIOStreamTransport(in)));
     } catch (org.apache.thrift.TException te) {
-      throw new java.io.IOException(te);
+      throw new java.io.IOException(te.getMessage());
     }
   }
 
diff -Naur gen-java/edu/berkeley/cs/amplab/carat/thrift/HogsBugs.java thrift/edu/berkeley/cs/amplab/carat/thrift/HogsBugs.java
--- gen-java/edu/berkeley/cs/amplab/carat/thrift/HogsBugs.java	2012-05-02 17:42:27.582948096 -0700
+++ thrift/edu/berkeley/cs/amplab/carat/thrift/HogsBugs.java	2012-05-02 21:32:08.547011150 -0700
@@ -24,8 +24,6 @@
 import java.util.BitSet;
 import java.nio.ByteBuffer;
 import java.util.Arrays;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 public class HogsBugs implements org.apache.thrift.TBase<HogsBugs, HogsBugs._Fields>, java.io.Serializable, Cloneable {
   private static final org.apache.thrift.protocol.TStruct STRUCT_DESC = new org.apache.thrift.protocol.TStruct("HogsBugs");
@@ -1039,7 +1037,7 @@
     try {
       write(new org.apache.thrift.protocol.TCompactProtocol(new org.apache.thrift.transport.TIOStreamTransport(out)));
     } catch (org.apache.thrift.TException te) {
-      throw new java.io.IOException(te);
+      throw new java.io.IOException(te.getMessage());
     }
   }
 
@@ -1049,7 +1047,7 @@
       __isset_bit_vector = new BitSet(1);
       read(new org.apache.thrift.protocol.TCompactProtocol(new org.apache.thrift.transport.TIOStreamTransport(in)));
     } catch (org.apache.thrift.TException te) {
-      throw new java.io.IOException(te);
+      throw new java.io.IOException(te.getMessage());
     }
   }
 
diff -Naur gen-java/edu/berkeley/cs/amplab/carat/thrift/NetworkDetails.java thrift/edu/berkeley/cs/amplab/carat/thrift/NetworkDetails.java
--- gen-java/edu/berkeley/cs/amplab/carat/thrift/NetworkDetails.java	2012-05-02 17:42:27.550947939 -0700
+++ thrift/edu/berkeley/cs/amplab/carat/thrift/NetworkDetails.java	2012-05-02 21:32:08.543011135 -0700
@@ -24,8 +24,6 @@
 import java.util.BitSet;
 import java.nio.ByteBuffer;
 import java.util.Arrays;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 public class NetworkDetails implements org.apache.thrift.TBase<NetworkDetails, NetworkDetails._Fields>, java.io.Serializable, Cloneable {
   private static final org.apache.thrift.protocol.TStruct STRUCT_DESC = new org.apache.thrift.protocol.TStruct("NetworkDetails");
@@ -805,7 +803,7 @@
     try {
       write(new org.apache.thrift.protocol.TCompactProtocol(new org.apache.thrift.transport.TIOStreamTransport(out)));
     } catch (org.apache.thrift.TException te) {
-      throw new java.io.IOException(te);
+      throw new java.io.IOException(te.getMessage());
     }
   }
 
@@ -815,7 +813,7 @@
       __isset_bit_vector = new BitSet(1);
       read(new org.apache.thrift.protocol.TCompactProtocol(new org.apache.thrift.transport.TIOStreamTransport(in)));
     } catch (org.apache.thrift.TException te) {
-      throw new java.io.IOException(te);
+      throw new java.io.IOException(te.getMessage());
     }
   }
 
diff -Naur gen-java/edu/berkeley/cs/amplab/carat/thrift/ProcessInfo.java thrift/edu/berkeley/cs/amplab/carat/thrift/ProcessInfo.java
--- gen-java/edu/berkeley/cs/amplab/carat/thrift/ProcessInfo.java	2012-05-02 17:42:27.546947923 -0700
+++ thrift/edu/berkeley/cs/amplab/carat/thrift/ProcessInfo.java	2012-05-02 21:32:08.547011150 -0700
@@ -24,8 +24,6 @@
 import java.util.BitSet;
 import java.nio.ByteBuffer;
 import java.util.Arrays;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 public class ProcessInfo implements org.apache.thrift.TBase<ProcessInfo, ProcessInfo._Fields>, java.io.Serializable, Cloneable {
   private static final org.apache.thrift.protocol.TStruct STRUCT_DESC = new org.apache.thrift.protocol.TStruct("ProcessInfo");
@@ -579,7 +577,7 @@
     try {
       write(new org.apache.thrift.protocol.TCompactProtocol(new org.apache.thrift.transport.TIOStreamTransport(out)));
     } catch (org.apache.thrift.TException te) {
-      throw new java.io.IOException(te);
+      throw new java.io.IOException(te.getMessage());
     }
   }
 
@@ -589,7 +587,7 @@
       __isset_bit_vector = new BitSet(1);
       read(new org.apache.thrift.protocol.TCompactProtocol(new org.apache.thrift.transport.TIOStreamTransport(in)));
     } catch (org.apache.thrift.TException te) {
-      throw new java.io.IOException(te);
+      throw new java.io.IOException(te.getMessage());
     }
   }
 
diff -Naur gen-java/edu/berkeley/cs/amplab/carat/thrift/Registration.java thrift/edu/berkeley/cs/amplab/carat/thrift/Registration.java
--- gen-java/edu/berkeley/cs/amplab/carat/thrift/Registration.java	2012-05-02 17:42:27.542947899 -0700
+++ thrift/edu/berkeley/cs/amplab/carat/thrift/Registration.java	2012-05-02 21:32:08.543011135 -0700
@@ -24,8 +24,6 @@
 import java.util.BitSet;
 import java.nio.ByteBuffer;
 import java.util.Arrays;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 public class Registration implements org.apache.thrift.TBase<Registration, Registration._Fields>, java.io.Serializable, Cloneable {
   private static final org.apache.thrift.protocol.TStruct STRUCT_DESC = new org.apache.thrift.protocol.TStruct("Registration");
@@ -669,7 +667,7 @@
     try {
       write(new org.apache.thrift.protocol.TCompactProtocol(new org.apache.thrift.transport.TIOStreamTransport(out)));
     } catch (org.apache.thrift.TException te) {
-      throw new java.io.IOException(te);
+      throw new java.io.IOException(te.getMessage());
     }
   }
 
@@ -679,7 +677,7 @@
       __isset_bit_vector = new BitSet(1);
       read(new org.apache.thrift.protocol.TCompactProtocol(new org.apache.thrift.transport.TIOStreamTransport(in)));
     } catch (org.apache.thrift.TException te) {
-      throw new java.io.IOException(te);
+      throw new java.io.IOException(te.getMessage());
     }
   }
 
diff -Naur gen-java/edu/berkeley/cs/amplab/carat/thrift/Reports.java thrift/edu/berkeley/cs/amplab/carat/thrift/Reports.java
--- gen-java/edu/berkeley/cs/amplab/carat/thrift/Reports.java	2012-05-02 17:42:27.574948056 -0700
+++ thrift/edu/berkeley/cs/amplab/carat/thrift/Reports.java	2012-05-02 21:32:08.543011135 -0700
@@ -24,8 +24,6 @@
 import java.util.BitSet;
 import java.nio.ByteBuffer;
 import java.util.Arrays;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 public class Reports implements org.apache.thrift.TBase<Reports, Reports._Fields>, java.io.Serializable, Cloneable {
   private static final org.apache.thrift.protocol.TStruct STRUCT_DESC = new org.apache.thrift.protocol.TStruct("Reports");
@@ -1036,7 +1034,7 @@
     try {
       write(new org.apache.thrift.protocol.TCompactProtocol(new org.apache.thrift.transport.TIOStreamTransport(out)));
     } catch (org.apache.thrift.TException te) {
-      throw new java.io.IOException(te);
+      throw new java.io.IOException(te.getMessage());
     }
   }
 
@@ -1046,7 +1044,7 @@
       __isset_bit_vector = new BitSet(1);
       read(new org.apache.thrift.protocol.TCompactProtocol(new org.apache.thrift.transport.TIOStreamTransport(in)));
     } catch (org.apache.thrift.TException te) {
-      throw new java.io.IOException(te);
+      throw new java.io.IOException(te.getMessage());
     }
   }
 
diff -Naur gen-java/edu/berkeley/cs/amplab/carat/thrift/Sample.java thrift/edu/berkeley/cs/amplab/carat/thrift/Sample.java
--- gen-java/edu/berkeley/cs/amplab/carat/thrift/Sample.java	2012-05-02 17:42:27.566948016 -0700
+++ thrift/edu/berkeley/cs/amplab/carat/thrift/Sample.java	2012-05-02 21:32:08.547011150 -0700
@@ -22,10 +22,9 @@
 import java.util.EnumSet;
 import java.util.Collections;
 import java.util.BitSet;
+import java.io.Serializable;
 import java.nio.ByteBuffer;
 import java.util.Arrays;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 public class Sample implements org.apache.thrift.TBase<Sample, Sample._Fields>, java.io.Serializable, Cloneable {
   private static final org.apache.thrift.protocol.TStruct STRUCT_DESC = new org.apache.thrift.protocol.TStruct("Sample");
@@ -1665,7 +1664,7 @@
     try {
       write(new org.apache.thrift.protocol.TCompactProtocol(new org.apache.thrift.transport.TIOStreamTransport(out)));
     } catch (org.apache.thrift.TException te) {
-      throw new java.io.IOException(te);
+      throw new java.io.IOException(te.getMessage());
     }
   }
 
@@ -1675,7 +1674,7 @@
       __isset_bit_vector = new BitSet(1);
       read(new org.apache.thrift.protocol.TCompactProtocol(new org.apache.thrift.transport.TIOStreamTransport(in)));
     } catch (org.apache.thrift.TException te) {
-      throw new java.io.IOException(te);
+      throw new java.io.IOException(te.getMessage());
     }
   }
 
